src/widgets/dnshole/DnsHoleControls.tsx

```diff
@@ -69,6 +81,12 @@ const dnsLightStatus = (
 
 function DnsHoleControlsWidgetTile({ widget }: DnsHoleControlsWidgetProps) {
   const { data: sessionData } = useSession();
+  const [opened, { close, open }] = useDisclosure(false);
+  const [hours, setHours] = useState(0);
+  const [minutes, setMinutes] = useState(0);
+  const hoursHandlers = useRef<NumberInputHandlers>();
+  const minutesHandlers = useRef<NumberInputHandlers>();
+  const [appId, setAppId] = useState('');
```

Comment:
Would be nice if you moved this to a new component to not clutter the main component. The same would apply to the other buttons but this is out of scope